plugins/outputs/file/file.go

```diff
@@ -39,13 +43,29 @@ func (f *File) SetSerializer(serializer serializers.Serializer) {
 	f.serializer = serializer
 }
 
-func (f *File) Connect() error {
-	writers := []io.Writer{}
-
+func (f *File) Init() error {
+	var err error
 	if len(f.Files) == 0 {
 		f.Files = []string{"stdout"}
 	}
 
+	var options []internal.EncodingOption
+	if f.CompressionLevel >= 0 && f.CompressionAlgorithm != "" && f.CompressionAlgorithm != "identity" {
+		options = append(options, internal.WithCompressionLevel(f.CompressionLevel))
+		f.encoder, err = internal.NewContentEncoder(f.CompressionAlgorithm, options...)
+	} else {
+		f.encoder, err = internal.NewContentEncoder(f.CompressionAlgorithm)
+	}
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
This can be simplified?
```suggestion
	return err
```